lib/api/v3/work_packages/schema/work_package_schemas_api.rb

```diff
@@ -0,0 +1,85 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module API
+  module V3
+    module WorkPackages
+      module Schema
+        class WorkPackageSchemasAPI < Grape::API
+          resources :schemas do
+
+            params do
+              requires :id, desc: 'Work package schema id'
+            end
+
+            # The schema id is an artificial identifier that is composed of a work packages
+            # project and its type (separated by a dash)
+            # This allows to have a separate schema URL for each kind of different work packages
+            # but with better caching capabilities than simply using the work package id as
+            # identifier for the schema
+            namespace ':id' do
```

Comment:
I already wanted to answer that this is not possible, because I could not find anything about this feature here: https://github.com/intridea/grape

However, I decided to at least try your proposal, to be sure that it does not work. Apparently it actually **does** work.

However, I seem to get trouble for URLs like `schemas/foo` now... Will have to investigate here...